MeshLib/MeshSubset.h

```diff
@@ -29,22 +29,31 @@ class MeshSubset
 {
 public:
     /// construct from nodes
-    MeshSubset(const Mesh& msh, std::vector<Node*> const& vec_items)
-        : _msh(msh), _nodes(&vec_items), _eles(nullptr)
+    MeshSubset(const Mesh& msh, std::vector<Node*> const& vec_items,
+        bool const delete_ptr = false)
+        : _msh(msh), _nodes(&vec_items), _eles(nullptr), _delete_ptr(delete_ptr)
     {}
 
     /// construct from elements
-    MeshSubset(const Mesh& msh, std::vector<Element*> const& vec_items)
-        : _msh(msh), _nodes(nullptr), _eles(&vec_items)
+    MeshSubset(const Mesh& msh, std::vector<Element*> const& vec_items,
+        bool const delete_ptr = false)
+        : _msh(msh), _nodes(nullptr), _eles(&vec_items), _delete_ptr(delete_ptr)
     {}
 
     /// construct from both nodes and elements
     MeshSubset(const Mesh& msh, std::vector<Node*> const& vec_nodes,
-               std::vector<Element*> const& vec_eles)
-        : _msh(msh), _nodes(&vec_nodes), _eles(&vec_eles)
+        std::vector<Element*> const& vec_eles, bool const delete_ptr = false)
+        : _msh(msh), _nodes(&vec_nodes), _eles(&vec_eles), _delete_ptr(delete_ptr)
     {}
 
-    ~MeshSubset() {}
+    ~MeshSubset()
+    {
+        if (_delete_ptr)
+        {
+            delete _nodes;
+            delete _eles;
+        }
+    }
```

Comment:
I think the whole ```delete_ptr``` solution is very unintuitive. At the very least it should be well-documented in constructor/destructor why this variable is there and its effect is